code/go/0chain.net/miner/protocol_block.go

```diff
@@ -932,6 +1067,19 @@ func (mc *Chain) generateBlock(ctx context.Context, b *block.Block,
 		return err
 	}
 
+	var costs []int
+	cost := 0
+	for _, txn := range b.Txns {
+		c, err := mc.EstimateTransactionCost(ctx, lfb, lfb.ClientState, txn)
```

Comment:
Isn't the  cost collected in `iterInfo`, why do we need to go through the txns again to recalculate. If the `processTxn` after the transaction iteration have cost, maybe we can estimate after each call instead of redo all.